### ISSUE TITLE
Create a cooldown for sound effects

### ISSUE THREAD

**Issue by jessemillar** [jessemillar/screm#26]
I’m not sure if I want a global cooldown or a cooldown per user. Probably global...? Might be more useful and also easier to implement.

- [ ] Implement global cooldown
- [ ] Put cooldown duration in chat message as suggested below

**Comment by hey24sheep** [comment]
Hi, thank you for this wonderful plugin. I have a few more ideas I will write them here so you can see if they are fit to be separate or not.

- Can we add the ability to get hotkey for quit or trigger from config too? 
- Can we have a check like only followers (like onlyFollowerMode or onlySubMode) can do !sfx but all users including unfollowed will see the welcome message so they can be tempted to follow ?
- This cooldown should be added to the chat message for everyone to see? like ` Playing a "blasting-off" sound for Hey24sheep!` should be ` Playing a "blasting-off" sound for Hey24sheep! Cooldown : 1min `. Now next command shows `Cooldown remaining : 30sec`?

I think global will be better than per user. As I saw a bug where all commands queued up instead of being played and they all played at once. Idk, how it happened.

**Comment by jessemillar** [comment]
@hey24sheep Thanks for your comment! It's exciting to know someone's actually played around with Screm!

- I created https://github.com/bfroggio/screm/issues/27 to track adding a quit shortcut. I originally had that implemented but removed it because I didn't have a personal use case for it. Out of curiosity, what's your use case? Do you just want a more convenient method for quitting when you're done streaming?
- Having a config for follower and sub mode is a GREAT idea. My channel is currently small enough that most people who watch follow but there's definitely a use case for larger channels where that filter would be super useful. Created https://github.com/bfroggio/screm/issues/28 to track.
- Very good idea adding the cooldown to the chat message. Will definitely do that.

Have you been able to reproduce the queue up bug you mentioned? I wonder if it was a Twitch problem since I've had a similar issue recently with my Twitch chat client (https://chatty.github.io/) not notifying me of new chat messages until thirty seconds later and then it notifies me of multiple messages at a time.

**Comment by hey24sheep** [comment]
> Thanks for your comment! It's exciting to know someone's actually played around with Screm!

Yes, I have been fiddling around with a lot of twitch APIs and stuff. So I found yours and it is quite good and different.

> Out of curiosity, what's your use case? Do you just want a more convenient method for quitting when you're done streaming?

I do not use the shortcuts for anything as of now. I just do manual quit only after I see my stream is done and successfully ended. But, I do use shortcuts for sfx playing. 

> Have you been able to reproduce the queue up bug you mentioned?

No, I have not been able to reproduce it. It is I think is from console getting blocked. What happens is, it just gets logged. Like, buffer is full but won't print. You have to manually go and press enter which runs everything queued up in the buffer and clears for new stuff and then it works fine for an hour or so.

>Having a config for follower and sub mode is a GREAT idea. My channel is currently small enough that most people who watch follow but there's definitely a use case for larger channels where that filter would be super useful.

My channel is super small like 10 followers so it won't help a ton but for follower only !sfx. It might help as in small streamers can reward new followers for at least something. Instead we make the chat follow mode.

Also, I found a new bug the volume scripts are not working. The volume is too loud for sfx only even on my desktop volume on 40ish.

**Comment by jessemillar** [comment]
@hey24sheep Thanks for the reply!

> No, I have not been able to reproduce it. It is I think is from console getting blocked. What happens is, it just gets logged. Like, buffer is full but won't print. You have to manually go and press enter which runs everything queued up in the buffer and clears for new stuff and then it works fine for an hour or so.

Hmmm. I'll see if I can reproduce this myself. Odd because the console shouldn't be interactive (I don't think). Please let me know if you hit this again.

> Also, I found a new bug the volume scripts are not working. The volume is too loud for sfx only even on my desktop volume on 40ish.

Do you mean that all sound effects are too loud? Or only some?

**Comment by hey24sheep** [comment]
All sound effects are too loud. I can't decrease the volume as the scripts aren't working.

I haven't got the bug again. It is not interactive it is like sometimes when logs are big and console gets stuck and you have to press enter to clear the buffer. It's that bug.

**Comment by jessemillar** [comment]
@hey24sheep I believe you have to have ffmpeg in your PATH for the scripts to work (if you didn't already see that). Regardless, I'm slowly working on a way to normalize the volume of all the sound effects using a compressor, but while that's in development, you can use "Sound mixer options" in Windows Settings to lower the overall volume of screm to be more to your liking.

**Comment by hey24sheep** [comment]
> I believe you have to have ffmpeg in your PATH for the scripts to work (if you didn't already see that).

No, I didn't it is not in the faq's or anything. I will check it.

> I'm slowly working on a way to normalize the volume of all the sound effects using a compressor, but while that's in development, you can use "Sound mixer options" in Windows Settings to lower the overall volume of screm to be more to your liking.

Yes, right now I do this.

**Comment by jessemillar** [comment]
@hey24sheep Good point on the scripts not being documented. Created a new issue to track that. https://github.com/bfroggio/screm/issues/29
